src/main/java/com/suse/saltstack/netapi/datatypes/Job.java

```diff
@@ -1,17 +1,68 @@
 package com.suse.saltstack.netapi.datatypes;
 
+import com.google.gson.annotations.SerializedName;
 import java.util.List;
 
+// todo - handle dates too (they are represented in really
+// exotic format: (2015, Mar 04 19:28:29.724698)
+
+/**
+ * Representation of a previously run job.
+ */
 public class Job {
 
-    private String jid;
-    private List<String> minions;
+    @SerializedName("Function")
+    private String function;
+
+    @SerializedName("Target")
+    private String target;
+
+    @SerializedName("Target-type")
+    private String targetType;
+
+    @SerializedName("User")
+    private String user;
+
+    @SerializedName("Arguments")
+    private List<String> arguments;
+
+    public String getFunction() {
+        return function;
+    }
+
+    public void setFunction(String function) {
```

Comment:
I agree with that @lucidd, we don't need the setters IIRC so let's have them removed.